### PR TITLE
Frontend user service

### LINKED ISSUE #18: Implement change username/password frontend
- [ ] Change username
- [ ] Change password

### PR DESCRIPTION
**Pull request by drake25122000**
Resolves #18 #19 #20 #21

### CODE REVIEW

**Review by florenciamartina — APPROVED** [review]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 /user-service/.env
+.env
 .idea/
-node_modules
\ No newline at end of file
+node_modules
+package-lock.json
+/*/package-lock.json
+/user-service/dump.rdb
```
Comment (erinmayg) [inline .gitignore:7]: no newline at EOF

File: frontend/.env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FAST_REFRESH=false
+CHOKIDAR_USEPOLLING=true
```
Comment (erinmayg) [inline frontend/.env:2]: remove .env from commit

File: frontend/src/@types/UserContext.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export interface User {
+    username: string | null;
+}
+  
+export interface UserContextInterface {
+    user: User;
+    login: (username: string, password: string) => {};
+}
```
Comment (erinmayg) [inline frontend/src/@types/UserContext.d.ts:8]: no newline at EOF

File: frontend/src/components/HomePage.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    const user = useUser();
+
+    return (
+        <div>
+            SUCCESSFULLY LOGGED {user.username} IN
+        </div>
+    );
+}
+
+export default HomePage;
```
Comment (erinmayg) [inline frontend/src/components/HomePage.tsx:15]: no newline at EOF

File: frontend/src/components/LoginPage.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            try {
+                await authClient.loginWithToken();
+                navigate("/home");
+            } catch (error) {
+                console.log("login " , error)
+            }
+        }
+
+        login();
+
```
Comment (erinmayg) [inline frontend/src/components/LoginPage.tsx:40]: remove blank line

File: frontend/src/components/modal/ChangePasswordDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import { URL_USER_CHANGE_PASSWORD } from "../../configs";
+import axios from "axios";
+import { changePassword } from "../../utils/auth-client";
+
+export default function ChangePasswordDialog({ isOpen }: { isOpen: boolean }) {
+
```
Comment (erinmayg) [inline frontend/src/components/modal/ChangePasswordDialog.tsx:13]: remove blank line

File: user-service/controller/user-controller.js [hunk 7]
```diff
@@ -13,13 +21,17 @@ export async function createUser(req, res) {
...
+      console.log("response controller: ");
       console.log(resp);
       if (resp.err) {
+        if (resp.err.name && resp.err.name === "MongoServerError" && resp.err.code === 11000) {
+          return res.status(409).json({ message: "Duplicate user!" });
+        }
         return res
           .status(400)
           .json({ message: "Could not create a new user!" });
+        // }
```
Comment (erinmayg) [inline user-service/controller/user-controller.js:34]: remove comment

File: user-service/controller/user-controller.js [hunk 8]
```diff
@@ -73,8 +76,212 @@ export async function signIn(req, res) {
...
+  });
+
+  await redisClient.connect();
+}
+
+export async function loginWithToken(req, res) {
+  try {
+    const { username } = req.body;
+    const token = req.headers.authorization.split(" ")[1];
+
```
Comment (erinmayg) [inline user-service/controller/user-controller.js:194]: remove extra blank line

**Review by erinmayg — COMMENTED** [review]
Minor changes, also for frontend, can run:
```npx prettier --write src```
to update coding style for unstaged files

File: frontend/src/App.tsx [hunk 9]
```diff
@@ -1,21 +1,53 @@
...
-                </Router>
-            </Box>
-        </div>
-    );
+  
+  const user = useUser();
+
+  return (
+    <div className="App">
+        <Box display={"flex"} flexDirection={"column"} padding={"4rem"}>
```
Comment (jzwoo) [inline frontend/src/App.tsx:22]: I think its better to add the stylings into the sx prop that material UI provides: https://mui.com/material-ui/react-box/

File: frontend/src/components/SignupPage.tsx [hunk 10]
```diff
@@ -1,99 +1,104 @@
...
-                type="password"
-                value={password}
-                onChange={(e) => setPassword(e.target.value)}
-                sx={{marginBottom: "2rem"}}
-            />
-            <Box display={"flex"} flexDirection={"row"} justifyContent={"flex-end"}>
-                <Button variant={"outlined"} onClick={handleSignup}>Sign up</Button>
-            </Box>
+  return (
+    <Box display={"flex"} flexDirection={"column"} width={"30%"}>
```
Comment (jzwoo) [inline frontend/src/components/SignupPage.tsx:59]: Stylings in sx prop

**Review by jzwoo — CHANGES_REQUESTED** [review]
If possible, try to break the MR into smaller portions, this MR is a little too big

**Review by erinmayg — APPROVED** [review]
LGTM

**Review by jzwoo — APPROVED** [review]
LGTM

**Pull request merged**